src/client/content_cao.cpp

```diff
@@ -826,11 +826,11 @@ void GenericCAO::setNodeLight(u8 light)
 	video::SColor color(255, light, light, light);
 
 	if (m_prop.visual == "wielditem" || m_prop.visual == "item") {
-		// Since these types of visuals are using their own shader
-		// they should be handled separately
-		if (m_wield_meshnode)
-			m_wield_meshnode->setColor(color);
-	} else if (m_enable_shaders) {
+		m_wield_meshnode->setNodeLightColor(color);
```

Comment:
`if (m_wield_meshnode)` missing.

Reply:
This code should also be moved into the `if (m_enable_shaders)` / `else` below.

Reply:
Oh, I wasn't expecting that.

@sfan5 no, setNodeLightColor handles m_enable_shaders by itself. Given that wield items setup materials by themselves I decided play along and do it this way.